src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -4006,65 +4733,109 @@ def _check_running(self):
         action = 0
 
         for cu in self._cus_to_watch:
+            #-------------------------------------------------------------------
+            # This code snippet reads the YARN application report file and if
+            # the application is RUNNING it update the state of the CU with the
+            # right time stamp. In any other case it works as it was.
+            if self._task_launcher.name == 'LaunchMethodYARN' \
```

Comment:
Same here: we should rethink our abstractions if Yarn requires so massive changes to the spawner.  Too much code redundency, and too much dependency no the LaunchMethod.  Maybe an evaluation hook on the launcher would help?  Like, `lm.create_command()`  before spawning, `lm.evaluate_result()` after collecting the job?